js/reference.js

```diff
@@ -63,12 +68,27 @@ var ERMrest = (function(module) {
             // validate the model references in the `uri` parameter
             // this method needs to internally construct a reference object that
             // represents the `uri` parameter
-            notimplemented();
+
+            // build reference
+            var reference = new Reference(location.href, location);
+            var server = this.getServer(reference._serviceUrl);
+            server.catalogs.get(reference._catalogId).then(function success(catalog) {
+                // Should I make a setter here?
+                reference.catalog = catalog;
```

Comment:
no, i don't think we need a setter. but i would make this declared as a private property `reference._catalog` instead